docs/src/main/asciidoc/build-items.adoc

```diff
@@ -0,0 +1,842 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+
+= Quarkus - Build Items
+
+Here you can find a list of Build Items and the extension that provides them:
+
+== Agroal - Database connection pool
+[%header,cols=2*]
+|===
+|Class Name |Description
+| https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/spi/src/main/java/io/quarkus/agroal/spi/JdbcDriverBuildItem.java[`io.quarkus.agroal.spi.JdbcDriverBuildItem`, window="_blank"]
+| +++Register a JDBC driver for the Agroal extension. <p> It allows to resolve automatically the driver from the kind, thus users don't have to set the driver anymore, except if they want to use a specific one.+++
+| https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/spi/src/main/java/io/quarkus/agroal/spi/JdbcDataSourceBuildItem.java[`io.quarkus.agroal.spi.JdbcDataSourceBuildItem`, window="_blank"]
+| +++A build item for JDBC datasources. <p> If you inject this build item when recording runtime init template calls, you are guaranteed the datasources configuration has been injected and datasources can be created.+++
+| https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/spi/src/main/java/io/quarkus/agroal/spi/JdbcDataSourceSchemaReadyBuildItem.java[`io.quarkus.agroal.spi.JdbcDataSourceSchemaReadyBuildItem`, window="_blank"]
+| +++A build item which can be used to order build processors which need a datasource's schema to be ready (which really means that the tables have been created and any migration run on them) for processing.+++
+| https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AggregatedDataSourceBuildTimeConfigBuildItem.java[`io.quarkus.agroal.deployment.AggregatedDataSourceBuildTimeConfigBuildItem`, window="_blank"]
+| +++<i>No Javadoc found</i>+++
```

Comment:
I wonder what can we do with the undocumented items and what can we do about the quality of the existing items? Perhaps as the first step, we should define some guidelines for documenting the BuildItems.

Reply:
Yes, something needs to be added to the extension author guide about this.